Library/ViewModels/AddNewCardViewModelTests.swift

```diff
@@ -340,41 +329,43 @@ internal final class AddNewCardViewModelTests: TestCase {
 
   func testUnsupportedCardMessage_showsWithInvalidCardBrand_AndExistingCardNumber() {
     self.vm.inputs.viewDidLoad()
+    self.vm.inputs.configure(with: .pledge, project: Project.template)
 
     self.creditCardValidationErrorContainerHidden
       .assertValues([true], "Unsupported card message is hidden on viewDidLoad")
 
-    self.vm.inputs.cardBrand(isValid: false)
-    self.vm.inputs.creditCardChanged(cardDetails: ("123", nil, nil, nil))
+    self.vm.inputs.creditCardChanged(cardDetails: ("123", nil, nil, nil, .generic))
 
     self.creditCardValidationErrorContainerHidden
       .assertValues([true, false], "Unsupported card message shows")
   }
 
   func testUnsupportedCardMessage_hidesWithValidCardBrand_AndExistingCardNumber() {
+    let project = Project.template
```

Comment:
We need to test `unsupportedCardMessage` for both the `.pledge` intent and the `.settings` intent, since they have slightly different logic.